### ISSUE TITLE
Add bufferfile() function for reading files into a memory buffer

### ISSUE THREAD

**Issue by RPGHacker** [RPGHacker/asar#16]
Add a bufferfile() function which buffers an arbitrary number of bytes from a file into a memory buffer. Subsequent readfile() functions on the same file will then try to read from this memory buffer before reading any data from file. Useful when doing file I/O on larger files, might lead to performance boosts.

**Comment by randomdude999** [comment]
Functions are supposed to take some inputs and return a double, what would this return then? Also, if you just wanted to buffer something without doing anything with the return value, you would need to use some hacks to ignore the value (such as `!define #= bufferfile() : undef !define`). IMO this would better be implemented as a command.

**Comment by RPGHacker** [comment]
Good point, I completely forgot that functions in Asar worked like this.

An alternate idea would be to have all file read functions secretly do some buffering under the hood (probably with a minimum and maximum buffer size) and then, whenever a readfile() was used, have it check if the data is already buffered. If not, free any previous buffers for that file and buffer the new data.

Example
canreadfile("file.dat", 0, 10000) <- Reads 0 to 10000 into the buffer for file file.dat
readfile1("file.dat", 100) <- Reads one byte from the buffer at position 100
readfile1("file.dat", 10100) <- Frees old buffer, reads x bytes from file.dat at position 10100 into buffer, reads one byte from the buffer at position 0

Something like that.

The idea for file buffering was really only to increase performance when doing a lot of reading on a single file, but honestly, I'm not sure if that's actually even needed. As far as I'm aware, operating systems already automatically cache file reads under the hood. I think they never read less than 4 KB from a file at any time, anyways, and I think cases where we ever need to read 4 KB or more with readfile() functions are probably quite rare already, so who knows, maybe this is a case of premature optimization to begin with.

(In any case, not sure why I added this to the milestone, since, if anything, this is really just a performance optimization).
